src/coordinated_workers/coordinator.py

```diff
@@ -785,6 +790,68 @@ def _reconcile_charm_labels(self) -> None:
             labels=self._coordinated_workers_solution_labels,
         )
 
+    def _get_mesh_policies_for_cluster_application(
+        self, source_model: str, source_application: str
+    ) -> List[MeshPolicy]:
+        """Return mesh policies that grant access for the specified cluster application to target all cluster units."""
+        cluster_models = self.cluster.gather_models()
+        mesh_policies: List[MeshPolicy] = []
+        for target_model in cluster_models:
+            mesh_policies.append(
+                MeshPolicy(
+                    source_namespace=source_model,
+                    source_app_name=source_application,
+                    target_namespace=target_model,
+                    target_selector_labels=self._coordinated_workers_solution_labels,
+                    target_type=PolicyTargetType.unit,
+                )
+            )
+        return mesh_policies
+
+    def _get_cluster_internal_mesh_policies(self) -> List[MeshPolicy]:
+        """Return all the required cluster internal mesh policies."""
+        mesh_policies: List[MeshPolicy] = []
+        # Coordinator -> Everything in the cluster
+        mesh_policies = self._get_mesh_policies_for_cluster_application(
+            self._charm.model.name, self._charm.app.name
+        )
+        # Workers -> Everything in the cluster
+        for worker_application in self.cluster.gather_applications():
```

Comment:
Not sure I agree with adding this new method.

Two alternatives:
1. have gather_topology deduplicate
`for w_app in self.cluster.gather_topology(per_app=True)`

2. do:
```
seen = set()
for w_app in self.cluster.gather_topology(per_app=True): 
    if w_app['application'] in seen: continue
    seen.add(w_app['application'])

```